Gordon360/Controllers/ProfilesController.cs

```diff
@@ -129,8 +129,23 @@ public ActionResult<string[]> GetCliftonStrengths_DEPRECATED(string username)
         {
             var id = _accountService.GetAccountByUsername(username).GordonID;
             var strengths = _profileService.GetCliftonStrengths(int.Parse(id));
+            if (strengths is null)
+            {
+                return Ok(Array.Empty<string>());
+            }
+            
+            var authenticatedUserName = AuthUtils.GetUsername(User);
+            if (string.IsNullOrEmpty(authenticatedUserName))
+            {
+                return strengths.Private is false 
+                    ? Ok(strengths.Themes) 
+                    : Ok(Array.Empty<string>());
+            }
 
-            return Ok(strengths?.Themes ?? Array.Empty<string>());
+            var authenticatedAccount = _accountService.GetAccountByUsername(authenticatedUserName);
+            return authenticatedAccount.GordonID == id 
+                ? Ok(strengths.Themes) 
+                : Ok(Array.Empty<string>());
```

Comment:
```
            var authenticatedUserName = AuthUtils.GetUsername(User);
            return strengths.Private is false || authenticatedUserName.CompareToIgnoreCase(username)
                ? Ok(strengths?.Themes) 
                : Ok(Array.Empty<string>());
```
This is the only change that needs to be made in the entire PR (for the two methods). If the strengths are public or if a user is querying their own profile, then return the strengths. If not, don't return the strengths.
Also add a using directive for `Gordon360.Extensions` at the top.